x-pack/plugins/security_solution/public/entity_analytics/components/asset_criticality/use_asset_criticality.ts

```diff
@@ -48,25 +49,46 @@ export const useAssetCriticalityData = ({
 }): State => {
   const QC = useQueryClient();
   const QUERY_KEY = [ASSET_CRITICALITY_KEY, entity.name];
-  const { fetchAssetCriticality, createAssetCriticality } = useEntityAnalyticsRoutes();
+  const { fetchAssetCriticality, createAssetCriticality, deleteAssetCriticality } =
+    useEntityAnalyticsRoutes();
 
   const privileges = useAssetCriticalityPrivileges(entity.name);
-  const query = useQuery<AssetCriticalityRecord, { body: { statusCode: number } }>({
+  const query = useQuery<AssetCriticalityRecord | null, { body: { statusCode: number } }>({
     queryKey: QUERY_KEY,
     queryFn: () => fetchAssetCriticality({ idField: `${entity.type}.name`, idValue: entity.name }),
     retry: (failureCount, error) => error.body.statusCode === 404 && failureCount > 0,
     enabled,
   });
 
-  const mutation = useMutation({
-    mutationFn: createAssetCriticality,
+  const mutation = useMutation<
+    AssetCriticalityRecord | DeleteAssetCriticalityResponse,
+    unknown,
+    Params,
+    unknown
+  >({
+    mutationFn: (params: Params) => {
+      if (params.criticalityLevel === 'unassigned') {
+        return deleteAssetCriticality({ idField: params.idField, idValue: params.idValue });
+      }
+
+      return createAssetCriticality({
+        idField: params.idField,
+        idValue: params.idValue,
+        criticalityLevel: params.criticalityLevel,
+      });
+    },
     onSuccess: (data) => {
-      QC.setQueryData(QUERY_KEY, data);
+      const queryData = 'deleted' in data ? null : data;
+      QC.setQueryData(QUERY_KEY, queryData);
     },
   });
 
```

Comment:
previously, we could just rely on whether the query 404'd to determine if the entity has no asset criticality. However now we have the ability to unassign. So I have set the query response body to null in that scenario (line 82( and added an extra check (87) to see if the asset criticality exists